### PR TITLE
add signature utility for display variable type.

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+     * <code>java.util.HashMap&lt;K,V&gt;</code> for
+     * <code>java.util.HashMap</code>.
+     * 
+     * @param type
+     *            the JDI type for evaluate
+     * @param showQuanlified
+     *            whether or not to show the fully qualified class name
+     * @return the display name
+     */
+    public static String getDisplayName(Type type, boolean showQuanlified) {
```
Comment (yaohaizh) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java:70]: typo qualified

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+import org.apache.commons.lang3.StringUtils;
+
+import com.sun.jdi.ClassLoaderReference;
+import com.sun.jdi.ReferenceType;
+import com.sun.jdi.Type;
+import com.sun.jdi.VirtualMachine;
+
+public abstract class SignatureUtils {
+    static final char ARRAY = '[';
+    static final char BYTE = 'B';
```
Comment (yaohaizh) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java:27]: org.eclipse.jdt.core.Signature has well defined constant values for these. Can we reuse that part?

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    private static String getTypeNameWithoutGeneric(String signature) {
+        int currentIndex = 0;
+        StringBuilder sb = new StringBuilder();
+        int arrayDimension = 0;
+        while (signature.charAt(arrayDimension) == ARRAY) {
+            arrayDimension++;
+        }
+
+        while (currentIndex < signature.length()) {
+            char key = signature.charAt(currentIndex++);
```
Comment (yaohaizh) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java:174]: Use map to simplify the operations here.

File: org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            ReferenceType componentType = (ReferenceType) iter.next();
+            ClassLoaderReference cl = componentType.classLoader();
+            if ((cl == null) ? (expectedClassLoader == null) : (cl.equals(expectedClassLoader))) {
+                return componentType;
+            }
+        }
+        return null;
+    }
+
+    private static String[] getTypeParameters(String typeSignature) throws IllegalArgumentException {
```
Comment (akaroml) [inline org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/SignatureUtils.java:119]: List&lt;String&gt;

**Review by akaroml — CHANGES_REQUESTED** [review]
As we discussed, let's investigate using JDT first. So please take down this pull request for now. Thanks.

**Pull request closed without merging**